src/tc-yaml.js

```diff
@@ -252,19 +252,24 @@ class VersionOne extends TcYaml {
       }
 
       config.tasks = config.tasks.map(task => {
-        task = _.defaults(task, {
+        task = Object.assign({
           taskId: defaultTaskId,
           taskGroupId: defaultTaskGroupId,
           created: now,
-        });
+          routes: config.reporting ? [cfg.app.checkTaskRoute] : [cfg.app.statusTaskRoute],
```

Comment:
What happens here if the user specifies routes on their task directly? Do these get overwritten?

I guess we'll have to say in our docs that you need to add specific routes to make up for the lost defaults if you have any yourself?

Reply:
Forgive me if this was already discussed.

Reply:
> What happens here if the user specifies routes on their task directly? Do these get overwritten?
I would assume so. Do we want to allow specifying routes? And if so - why? Do people currently specify routes? If so - why, without the bindings?

My understanding of the purpose of routes was that we need them to separate Checks from Statuses, or to have the pulse messages come through two separate channels. Once Statuses are deprecated and gone from the codebase, routes (and the bindings) will be gone as well.

This looks like a design-stage topic to me. I don't recall us discussing this during the rfc stage. I'm not against implementing that if we decide it's needed, but not in this PR.

Reply:
...unless you mean people would add routes to have something else listening to those. If that's the case - I can organize pushing into an array here. Let me know if that's what you meant (and I'm still curious about the use cases)

Reply:
Yeah, people add routes for things like [notifications](https://github.com/taskcluster/taskcluster-notify#taskcluster-notifications-service). I think pushing if something is already there makes sense! Unfortunately we have a limit (I think it is 10) to the number of routes you can add. Maybe there needs to be some sort of check that we won't go over that? For now just push and we can figure that out later.

Reply:
Oh that's actually awesome! I didn't realize we had that functionality. If there's a limit, something must have the check somewhere. Imma check it out

Reply:
[Found it](https://github.com/taskcluster/taskcluster-queue/blob/3b2960adf20ccb960a474fd68a526b4c389868b4/schemas/v1/task.yml#L94) - I would hope if people have 768 routes in their yml it would fail the schema check? 

UPD: [Checked that](https://github.com/owlishDeveloper/helloWorld/pull/93/files) - the schema checks happily passes

I would leave this for now (64 seems like a large enough number), but in a separate PR I would like to enforce task definition schema checks

Reply:
UPD2: [The schema check failed after all!](https://github.com/owlishDeveloper/helloWorld/pull/93) Yay. 

In a separate PR then I will deal with the bug of creating status when tasks were not submitted. This is not the first time I see it

Reply:
Added mention to the documentation for now.